main.go

```diff
@@ -78,6 +78,31 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	if len(flag.Args()) == 0 {
+		getBuf(os.Stdin, 0, nInvertParameter, quietParameter, maxCountParameter, regArr)
+	} else {
+		for i := range flag.Args() {
+			file, fileErr = os.Open(flag.Args()[i])
+			if fileErr != nil {
+				fmt.Fprintln(os.Stderr, fileErr)
+				os.Exit(2)
+			}
+
+			getBuf(file, i, nInvertParameter, quietParameter, maxCountParameter, regArr)
+			file.Close()
+		}
+	}
+
+	if counter <= 0 {
+		os.Exit(1)
+	}
+}
+
+func getBuf(file *os.File, u int, nInvertParameter *bool, quietParameter *bool, maxCountParameter *int, regArr []*regexp.Regexp) {
```

Comment:
Why are you passing the CLI params as pointers?

Reply:
Either I pass the CLI parameters as pointers or I pass the values to getBuf as pointers. It makes no difference does it?

Reply:
Why pointers? What does need `getBuf` **the pointers** for?

Reply:
The the CLI params! Not the file!